boa/src/builtins/string/mod.rs

```diff
@@ -1165,6 +1166,81 @@ impl String {
         }
     }
 
+    /// String.prototype.split()
+    ///
+    /// The `split()` method divides a String into an ordered list of substrings, puts these substrings into an array, and returns the array.
+    ///
+    /// The division is done by searching for a pattern; where the pattern is provided as the first parameter in the method's call.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.split
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split
+    pub(crate) fn split(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let string = this.to_string(context)?;
+
+        let separator = if args.is_empty() {
+            None
+        } else {
+            let arg0 = args
+                .get(0)
+                .expect("Failed to get argument for String method");
+
+            if arg0.is_null_or_undefined() {
+                None
+            } else {
+                if let Some(splitter) =
+                    arg0.get_property(context.well_known_symbols().split_symbol())
+                {
+                    let splitter = splitter.as_data_descriptor().unwrap().value();
+                    if splitter.is_function() {
+                        let arguments = vec![
+                            Value::from(string),
+                            args.get(1)
+                                .map(|x| x.to_owned())
+                                .unwrap_or(Value::Undefined),
+                        ];
+                        return context.call(&splitter, this, &arguments);
+                    } else if !splitter.is_undefined() {
+                        return Err(Value::from(
+                            "TypeError: separator[Symbol.split] is not a function",
+                        ));
+                    }
+                }
+                arg0.to_string(context).ok()
+            }
+        };
+
+        let limit = if args.len() < 2 {
+            std::u32::MAX as usize
+        } else {
+            args.get(1)
+                .expect("Could not get the limit argument")
+                .to_integer(context)? as usize
+        };
```

Comment:
This would be more idiomatic using:
```rust
args.get(1).map(|arg| arg.to_integer(context).map(|limit|limit as usize)).transpose()?.unwrap_or(std::u32::MAX as usize);
```
Or something like it